### DISCUSSION TITLE
Switching from Jekyll to Hugo

### DISCUSSION THREAD

**Discussion by glaslos** [honeynet/honeynet.github.io#13] [General]
Before we commit to Jekyll, I'd like to consider using Hugo instead. I'm not using Ruby for anything else, hence my struggles getting this to work while with Hugo I just download a binary and my dependencies are done. Should drastically lower the entry barrier.

**Comment by mhils** [comment]
Totally forgot about that part - that's a very good point in favor of Hugo. 👍

**Comment by xandfury** [comment]
But some will have the same argument about `go` 😄 Like why `go` and why not something like `rust` or `python`? Personally, I don't have a preference for any. Since I doubt a lot of `ruby` will be required for Jekyll itself. Afaik it should be mostly about markdown files. 

Another good option might be something like [Gatsby](https://github.com/gatsbyjs/gatsby) which allows integration from both markdown files and wordpress blogs - to make the transition even easier. Apparently, it would allow our static website to query the existing wp blog using GraphQL and the add static md pages on top of it. Not sure about how complicated this would be.. But sharing since we're talking about options.

**Reply by mhils** [reply]
The benefit of Hugo is that you download a single static precompiled binary, you do not need to set up a Go env. Any other compiled language would do as well. With Jekyll, you need to have passing understanding of how to work with Ruby/RubyGems.

**Reply by xandfury** [reply]
ah. Didn't know it was a static binary. That's reasonable (as long as they provide a binaries for major platforms; I am sure they do!). But since we're on the subject, do we know what benefits Hugo offer in comparison to Rust frameworks?

Do you folks think that we'll be writing some basic tests for the website? If yes, would you consider writing them in `go` or some other lang?

**Comment by xandfury** [comment]
@glaslos @mhils Just to be clear, I don't have any issues with Hugo if you think that would be more inviting for folks to contribute. I am a backend person, so someone needs to help me with the theme. Happy to take care of the rest. 

I just ran one of the Jekyll to Hugo scripts and pushed it to the repo. The theme is broken again.. I am more than happy to take care of the pages and the links. Do you all think we can buy a theme? Or have it custom made acc to our needs?

**Reply by xandfury** [reply]
Or maybe BoD and C level execs should talk design and document it somewhere? 😄 What kind of website do we want? What colours?  etc.. 
cc: @hgascon @aikinci

**Reply by mhils** [reply]
Let me try to crank out a simple hugo + template MVP today that we can then fill with contents. I feel the current template looks reasonably good, but is an awful mess implementation-wise. We can probably simplify that drastically and then go with that. :)

**Reply by mhils** [reply]
Here we go: https://honeynet.github.io/, https://github.com/honeynet/honeynet.github.io. I've built a new theme for Hugo to get rid of the insane amounts of cruft in the Wordpress template, and everything gets auto-deployed to GitHub pages. Now we "just" need to port over existing content 😅. I've added a TODO list to the README. :)
